orderedmap.go

```diff
@@ -5,9 +5,13 @@ type OrderedMap struct {
 	ll list
 }
 
-func NewOrderedMap() *OrderedMap {
+func NewOrderedMap(size ...int) *OrderedMap {
```

Comment:
I'm OK with this suggestion, but we shouldn't abuse the constructor like this. Create a `NewOrderedMapWIthCapacity` function.